packages/react-native/React/Fabric/Mounting/ComponentViews/Text/RCTParagraphComponentView.mm

```diff
@@ -27,22 +29,56 @@
 
 using namespace facebook::react;
 
+#if TARGET_OS_OSX // [macOS
+@interface RCTParagraphComponentUnfocusableTextView : NSTextView
+@end
+
+@implementation RCTParagraphComponentUnfocusableTextView
+
+- (BOOL)canBecomeKeyView
+{
+  return NO;
+}
+
+- (BOOL)resignFirstResponder
+{
+  // Don't relinquish first responder while selecting text.
+  if (self.selectable && NSRunLoop.currentRunLoop.currentMode == NSEventTrackingRunLoopMode) {
+    return NO;
+  }
+
+  return [super resignFirstResponder];
+}
+
+@end
+
+@interface RCTParagraphComponentView () <NSTextViewDelegate>
+@end
+#endif // macOS]
+
 #if !TARGET_OS_OSX // [macOS]
 @interface RCTParagraphComponentView () <UIEditMenuInteractionDelegate>
 
 @property (nonatomic, nullable) UIEditMenuInteraction *editMenuInteraction API_AVAILABLE(ios(16.0));
 
 @end
```

Comment:
Prefer the code blocks that are the same class to be in the same ifdef group. Something like:

```suggestion
#endif // macOS]

#if !TARGET_OS_OSX // [macOS]
@interface RCTParagraphComponentView () <UIEditMenuInteractionDelegate>

@property (nonatomic, nullable) UIEditMenuInteraction *editMenuInteraction API_AVAILABLE(ios(16.0));

@end
#else // [macOS
@interface RCTParagraphComponentView () <NSTextViewDelegate>
@end
#endif //macOS]
```